### PR TITLE
register solvers to data connector

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Allow the access to field solver, stage current background and current interpolation via the dataconnector.

This is a pre-PR to provide the support of creating fileds based on the particle current in the species initialization pipeline.

### CODE REVIEW

File: include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp [hunk 1]
```diff
@@ -95,14 +96,14 @@ namespace picongpu
                  * @param step index of time iteration
                  * @param fieldSolver field solver
                  */
-                void operator()(uint32_t const step, fields::Solver& fieldSolver) const
+                void operator()(uint32_t const step, fields::Solver& fieldSolver, bool allowCommunication) const
```
Comment (steindev) [inline include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp]: I don't see the reason to introduce this variable, as you set it to always `true` in `include/picongpu/simulation/control/Simulation.hpp` anyways.

**Review by steindev — DISMISSED** [review]
Only one minor comment.

File: include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp (hunk 1 above)
Comment (PrometheusPi) [reply include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp]: @psychocoderHPC Could you explain, why you want to have the option to disable communication?

**Review by PrometheusPi — DISMISSED** [review]
Just a question in agreement with @steindev

File: include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp]: I can remove this from this PR. I overlooked this change, it is required for the StaticCurrentDistribution feature.
For StaticCurrentDistribution we need to communicate the fields only once depending on the INterpolation stencil width.

Comment (psychocoderHPC) [reply include/picongpu/simulation/stage/CurrentInterpolationAndAdditionToEMF.hpp]: I removed the changes from this PR and will introduce it together with the new feature.

**Comment by psychocoderHPC** [comment]
ready to be merged!

**Review by steindev — APPROVED** [review]

**Pull request merged**